.github/actions/javascript/authorChecklist/authorChecklist.ts

```diff
@@ -0,0 +1,159 @@
+
+    if (numberOfFinishedChecklistItems >= minCompletedItems && numberOfUnfinishedChecklistItems === 0) {
+        console.log('PR Author checklist is complete 🎉');
+        return;
+    }
+
+    console.log(`Make sure you are using the most up to date checklist found here: ${pathToAuthorChecklist}`);
+    core.setFailed("PR Author Checklist is not completely filled out. Please check every box to verify you've thought about the item.");
+}
+
+async function generateDynamicChecksAndCheckForCompletion() {
+    // Generate dynamic checks
+    const checks = new Set<string>();
+    const categories = await getChecklistCategoriesForPullRequest();
+    for (const checksForCategory of categories) {
+        for (const check of checksForCategory) {
+            checks.add(check);
+        }
+    }
+
+    const body = github.context.payload.pull_request?.body ?? '';
+
+    // eslint-disable-next-line prefer-const
+    let [contentBeforeChecklist, checklist, contentAfterChecklist] = partitionWithChecklist(body);
+
+    let isPassing = true;
+    let checklistChanged = false;
+    for (const check of checks) {
+        // Check if it's already in the PR body, capturing the whether or not it's already checked
+        const regex = new RegExp(`- \\[([ x])] ${escapeRegExp(check)}`);
+        const match = regex.exec(checklist);
+        if (!match) {
+            // Add it to the PR body
+            isPassing = false;
+            checklist += `- [ ] ${check}\r\n`;
+            checklistChanged = true;
+        } else {
+            const isChecked = match[1] === 'x';
+            if (!isChecked) {
+                isPassing = false;
+            }
+        }
+    }
+    const allChecks = Object.values(CHECKLIST_CATEGORIES).reduce((acc: string[], category) => acc.concat(category.items), []);
+
+    for (const check of allChecks) {
+        if (!checks.has(check)) {
+            // Check if some dynamic check has been added with previous commit, but the check is not relevant anymore
+            const regex = new RegExp(`- \\[([ x])] ${escapeRegExp(check)}\r\n`);
+            const match = regex.exec(checklist);
+            if (match) {
+                // Remove it from the PR body
+                checklist = checklist.replace(match[0], '');
+                checklistChanged = true;
+            }
+        }
+    }
+
+    // Put the PR body back together, need to add the markers back in
+    const newBody = contentBeforeChecklist + checklistStartsWith + checklist + checklistEndsWith + contentAfterChecklist;
+
+    // Update the PR body
+    if (checklistChanged) {
+        await GithubUtils.octokit.pulls.update({
+            owner: CONST.GITHUB_OWNER,
+            repo: CONST.APP_REPO,
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            pull_number: prNumber,
+            body: newBody,
+        });
+        console.log('Updated PR checklist');
+    }
+
+    if (!isPassing) {
```

Comment:
I don't see why this block is needed

Reply:
This action now has two responsibilities:
1. Do dynamic checks and add new checks to PR description.
2. Check if all checkboxs are checked.

If we've added new checks to the description, we mark action as failed and we do not check second step.